source/adios2/toolkit/sst/cp/cp_writer.c

```diff
@@ -1470,6 +1470,14 @@ static void CP_PeerFailCloseWSReader(WS_ReaderInfo CP_WSR_Stream,
                 CP_WSR_Stream->Connections[i].CMconn = NULL;
             }
         }
+
+        if (CP_WSR_Stream->DP_WSR_Stream)
+        {
+            CP_WSR_Stream->ParentStream->DP_Interface->destroyWriterPerReader(
```

Comment:
@eisenhauer I noticed that `destroyWriterPerReader` was not called in any case, I added this here, seems to work well for the `mpi_dp` but I wonder if it would break other dp implementations